lib/src/main/java/com/knuddels/jtokkit/EncodingFactory.java

```diff
@@ -7,162 +7,169 @@
+     *
+     * @return an {@link Encoding} instance for the p50k_edit encoding
+     */
+    public static Encoding p50kEdit() {
+        return fromPredefinedParameters(
+                "p50k_edit",
+                "'(?:[sdmt]|ll|ve|re)| ?\\p{L}+| ?\\p{N}+| ?[^\\s\\p{L}\\p{N}]+|\\s+(?!\\S)|\\s+",
+                "/com/knuddels/jtokkit/p50k_base.tiktoken",
+                SPECIAL_TOKENS_P50K_EDIT,
+                false
+        );
+    }
+
+    /**
+     * Returns an {@link Encoding} instance for the cl100k_base encoding.
+     *
+     * @return an {@link Encoding} instance for the cl100k_base encoding
+     */
+    public static Encoding cl100kBase() {
+        return fromPredefinedParameters(
+                "cl100k_base",
+                "'(?:[sdmt]|ll|ve|re)|[^\r\n\\p{L}\\p{N}]?+\\p{L}+|\\p{N}{1,3}| ?[^\\s\\p{L}\\p{N}]++[\r\n]*|\\s*[\r\n]|\\s+(?!\\S)|\\s+",
+                "/com/knuddels/jtokkit/cl100k_base.tiktoken",
+                SPECIAL_TOKENS_CL100K_BASE,
+                true
+        );
+    }
+
+    /**
+     * Returns an {@link Encoding} instance for the given GPT BytePairEncoding parameters.
+     *
+     * @param parameters the GPT BytePairEncoding parameters
+     * @return an {@link Encoding} instance for the given GPT BytePairEncoding parameters
+     */
+    public static Encoding fromParameters(GptBytePairEncodingParams parameters) {
+        return new GptBytePairEncoding(parameters);
+    }
+
+    private static Encoding fromPredefinedParameters(
+            String name,
+            String patternString,
+            String fileName,
+            Map<String, Integer> specialTokens,
+            boolean caseInsensitive
+    ) {
+        Pattern regex = compileRegex(patternString, caseInsensitive);
+        Map<byte[], Integer> mergeableRanks = loadMergeableRanks(fileName);
+        GptBytePairEncodingParams params = new GptBytePairEncodingParams(name, regex, mergeableRanks, specialTokens);
+        return fromParameters(params);
+    }
+
+    static Pattern compileRegex(String patternString, boolean caseInsensitive) {
+        try {
+            int flags = Pattern.UNICODE_CHARACTER_CLASS;
+            if (caseInsensitive) {
+                flags |= Pattern.CASE_INSENSITIVE;
+            }
+            return Pattern.compile(patternString, flags);
+        } catch (IllegalArgumentException exception) {
+            // Workaround for Android where an IllegalArgumentException is thrown when using UNICODE_CHARACTER_CLASS
+            return Pattern.compile(patternString);
+        }
+    }
+
+    public static Map<byte[], Integer> loadMergeableRanks(String fileName) {
+        try (InputStream in = EncodingFactory.class.getResourceAsStream(fileName)) {
+            if (in == null) {
+                throw new IllegalStateException("Could not find " + fileName + " in resources");
+            }
+
+            Map<byte[], Integer> mergeableRanks = new LinkedHashMap<>(); // keep order to optimize collisions
```

Comment:
the order of the tokens represents how likely they are, so we want to preserve that order when inserting it into the `HashMap` so that when collisions occur, the more likely ones are in front of the list (or exact match in case of OpenAddressing map)